### PR TITLE
Add `RedisPoolStore` implementation 

### PR DESCRIPTION
**Pull request by FrancescoLuzzi**
This PR implements a new store `RedisPoolStore` using `redis_pool` (a wrapper over `redis`) following #96.

I thought of merging `RedisStore` and `RedisPoolStore` like you did with `sqlx-store`,
but coming from different crates I ultimately decided to separe them.

Let me know if I need to fix anything.

Cheers!

### CODE REVIEW

File: redis-pool-store/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    type Error = RedisStoreError;
+
+    async fn save(&self, session: &Session) -> Result<(), Self::Error> {
+        let expire = OffsetDateTime::unix_timestamp(session.expiry_date());
+        let mut con = self.client.aquire().await?;
+        redis::pipe()
+            .atomic() //makes this a transation.
+            .set(session.id().to_string(), rmp_serde::to_vec(&session)?)
+            .ignore()
+            .expire_at(session.id().to_string(), expire as usize)
```
Comment (maxcountryman) [inline redis-pool-store/src/lib.rs:67]: I think this isn't what was intended.

File: redis-pool-store/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+version.workspace = true
+license.workspace = true
+edition.workspace = true
+authors.workspace = true
+repository.workspace = true
+
+[dependencies]
+async-trait = { workspace = true }
+redis_pool = "0.2.1"
+redis = "0.23.4"
```
Comment (maxcountryman) [inline redis-pool-store/Cargo.toml:14]: I'm okay to support fred and redis but this is a confusing interface. Both stores are currently called "redis" but depend on different crates for the client.

Comment (FrancescoLuzzi) [reply redis-pool-store/Cargo.toml:14]: I agree, in fact I wanted to discuss the integration before investing too much time on it.

The exposed features needs to define which client they are using (eg. `redis_fred`) without being reduntant (eg. `redis_redis`).
Maybe renaming `RedisStore` to `RedisFredStore` could also help with clarity at the cost of compatibility.

All these thoughts come from the prospective of not merging the interfaces like `sqlx-store`,
but I'm open to reconsider my position😄

File: redis-pool-store/src/lib.rs (hunk 1 above)
Comment (FrancescoLuzzi) [reply redis-pool-store/src/lib.rs]: Hope this is what you intended c8291f60634b24bfbe40b44e400075f7a71e6ae3

File: redis-pool-store/Cargo.toml (hunk 2 above)
Comment (maxcountryman) [reply redis-pool-store/Cargo.toml]: I think we'll want to resolve https://github.com/maxcountryman/tower-sessions/issues/102 to help address this.

**Comment by maxcountryman** [comment]
We're beginning to break these implementations out into their own third party crates. Would you be interested in publishing this that way? I'd love to maintain a catalog of available crates in the README here so folks can more easily discover the various backend stores.

**Comment by FrancescoLuzzi** [comment]
I'd like to do it, but since I don't use this library very often I would just abandon it and not update it.

Also my choice of using `RedisPool` was not particularly great, `[rd2d](https://github.com/sfackler/r2d2)` would be much better.

Thanks anyway for your time and for the work you put in this library, have a good year!

Peace ✌

**Pull request closed without merging**